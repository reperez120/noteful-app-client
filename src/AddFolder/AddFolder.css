
.AddFolder {
  color: white;
}

/* .folderNameInput {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
} */

.addFolderButton {
  height: 3rem;
} 

.folderNameInput {
  width: 2rem;
}