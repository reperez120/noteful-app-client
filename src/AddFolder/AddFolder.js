import React, { Component } from 'react';
import NotefulForm from '../NotefulForm/NotefulForm';
import NotefulError from '../NotefulError';
import PropTypes from 'prop-types';
import './AddFolder.css';
import { withRouter } from 'react-router-dom';
import ApiContext from '../ApiContext';
import config from '../config';

class AddFolder extends Component {

  state = {
    error: null,
  };

  constructor(props) {
    super(props);
    this.state = {
      name: ''
    }
  }

  static contextType = ApiContext;

  handleFolderSubmit = e => {
    e.preventDefault()
    const {name} = e.target
    const folder = {
      name: name.value
    }
    const url = `${config.API_ENDPOINT}/api/folders`
    
    const options = {
    method: 'POST',
    headers: {
        'content-type': 'application/json'
    },
    body: JSON.stringify(folder),
    }
    this.setState({ error: null })
    fetch(url, options)
      .then(res => {
        if(!res.ok) {
          throw new Error('Something went wrong, please try again later');
        }
        return res.json();
      })
      .then((data) => {
        this.context.addFolder(data)
        this.props.history.push('/')
      })
      .catch(err => {
        this.setState({
          error: err.message
        });
      })
  }

  render() {
    return (
      <section className='AddFolder'>
        <NotefulError>
        <h2>Create a folder</h2>
        <NotefulForm onSubmit={this.handleFolderSubmit}>
          <div className='field'>
            <label htmlFor='name-input'>
              Name
            </label>
            <input type='text' id='name'/>
          </div>
          <div className='buttons'>
            <button type='submit' className="addFolderButton">
              Add folder
            </button>
          </div>
        </NotefulForm>
        </NotefulError>
      </section>
    )
  }
}

AddFolder.propTypes = {
  addFolder: PropTypes.func.isRequired,
  folders: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired
  })) 
};

export default withRouter(AddFolder);