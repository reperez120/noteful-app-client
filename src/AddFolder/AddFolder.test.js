import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddFolder from './AddFolder'

describe(`AddFolder component`, () => {
  it('renders the add folder form', () => {
    const wrapper = shallow(<AddFolder />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
