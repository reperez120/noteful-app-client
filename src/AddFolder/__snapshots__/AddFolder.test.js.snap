// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder component renders the add folder form 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`AddItemForm component renders the complete form 1`] = `
<section
  className="AddFolder"
>
  <h2>
    Create a folder
  </h2>
  <NotefulForm>
    <div
      className="field"
    >
      <label
        htmlFor="folder-name-input"
      >
        Name
      </label>
      <input
        id="folder-name-input"
        type="text"
      />
    </div>
    <div
      className="buttons"
    >
      <button
        type="submit"
      >
        Add folder
      </button>
    </div>
  </NotefulForm>
</section>
`;
